### ISSUE TITLE
AVX2 Optimization?

### ISSUE THREAD

**Issue by medmr1** [alicemaz/rust-base64#55]
Hi,

Just wondering: Would you be interested in tackling AVX2 optimization within this crate?
There's some info here:
https://lemire.me/blog/2018/01/17/ridiculously-fast-base64-encoding-and-decoding/

**Comment by marshallpierce** [comment]
Yes, I've thought about it. I've actually already ported some of Daniel Lemire &co's other work to Rust (with SIMD): https://bitbucket.org/marshallpierce/stream-vbyte-rust

I think it might be best to create a separate project for experimentation so that we can freely alter the API. This could also open up some possibilities to do things like allow streaming base64 processing more naturally.

**Comment by marshallpierce** [comment]
I made https://github.com/marshallpierce/rust-base64-experimental just to have a place holder to keep track of ideas, etc.
